src/ObjectUtils/include/ObjectUtils/ObjectFile.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef OBJECT_UTILS_OBJECT_FILE_H_
+#define OBJECT_UTILS_OBJECT_FILE_H_
+
+#include <llvm/Object/ObjectFile.h>
+
+#include <filesystem>
+#include <memory>
+#include <string>
+
+#include "OrbitBase/Result.h"
+#include "llvm/Object/Binary.h"
+#include "symbol.pb.h"
+
+namespace orbit_object_utils {
+
+class ElfFile;
+
+class ObjectFile {
+ public:
+  ObjectFile() = default;
+  virtual ~ObjectFile() = default;
+
+  [[nodiscard]] virtual ErrorMessageOr<orbit_grpc_protos::ModuleSymbols> LoadSymbols() = 0;
+  [[nodiscard]] virtual bool HasSymbols() const = 0;
+  [[nodiscard]] virtual std::string GetPrettyName() const = 0;
+  [[nodiscard]] virtual std::filesystem::path GetFilePath() const = 0;
+
+  [[nodiscard]] static ErrorMessageOr<std::unique_ptr<ElfFile>> CreateElfFile(
+      const std::filesystem::path& file_path);
+  [[nodiscard]] static ErrorMessageOr<std::unique_ptr<ElfFile>> CreateElfFile(
+      const std::filesystem::path& file_path,
+      llvm::object::OwningBinary<llvm::object::ObjectFile>&& file);
+  [[nodiscard]] static ErrorMessageOr<std::unique_ptr<ElfFile>> CreateElfFileFromBuffer(
+      const std::filesystem::path& file_path, const void* buf, size_t len);
```

Comment:
I think these should be free functions and declared in `ElfFile.h`. This would also avoid the forward declaration of `ElfFile`.